js/data/index.d.ts

```diff
@@ -11,18 +11,29 @@ export type SearchOperation = '=' | '<>' | '>' | '>=' | '<' | '<=' | 'startswith
 
 type KeySelector<T> = string | ((source: T) => string | number | Date | Object);
 
-type BaseGroupDescriptor<T> = {
+type SelectionDescriptor<T> = {
     selector: KeySelector<T>;
 };
 
+type OrderingDescriptor<T> = SelectionDescriptor<T> & {
+    desc?: boolean;
+};
+
+/**
+ * @docid
+ * @public
+ */
+export type GroupingInterval = 'year' | 'quarter' | 'month' | 'day' | 'dayOfWeek' | 'hour' | 'minute' | 'second';
+
 /**
  * @docid
  * @public
  * @type object
  * @skip
  */
-export type GroupDescriptor<T> = KeySelector<T> | BaseGroupDescriptor<T> & {
-    desc?: boolean;
+export type GroupDescriptor<T> = KeySelector<T> | OrderingDescriptor<T> & {
+    groupInterval?: number | GroupingInterval;
+    isExpanded?: boolean;
 };
```

Comment:
**Minor**
Adding braces should increase readability:
```suggestion
export type GroupDescriptor<T> = KeySelector<T> | (OrderingDescriptor<T> & {
    groupInterval?: number | GroupingInterval;
    isExpanded?: boolean;
});
```

Reply:
fixed